moveit_ros/moveit_servo/src/servo_calcs.cpp

```diff
@@ -207,13 +207,16 @@ void ServoCalcs::run(const ros::TimerEvent& timer_event)
   // Always update the joints and end-effector transform for 2 reasons:
   // 1) in case the getCommandFrameTransform() method is being used
   // 2) so the low-pass filters are up to date and don't cause a jump
-  while (!updateJoints() && ros::ok())
+  while (!updateJoints() && ros::ok() && !stop_requested_)
   {
-    if (stop_requested_)
-      return;
     default_sleep_rate_.sleep();
   }
 
+  if (stop_requested_)
+  {
+    return;
+  }
+
```

Comment:
Is the fix basically that if ros::ok() is false amd stop_requested_ is true this continues instead of returning?

Reply:
yep. Also, if updateJoints was successful immediately, there wasn't a check on stop_requested_